S3Source.h

```diff
@@ -0,0 +1,98 @@
+#if !defined(S3Source_h)
+#define S3Source_h
+
+#include <string>
+#include <memory>
+#include <chrono>
+#include <iostream>
+#include <utility>
+
+#include "SharedSourceBase.h"
+#include "DataProductRetriever.h"
+#include "DelayedProductRetriever.h"
+#include "SerialTaskQueue.h"
+#include "DeserializeStrategy.h"
+#include "S3Common.h"
+#include "objectstripe.pb.h"
+
+
+namespace cce::tf {
+class S3DelayedRetriever : public DelayedProductRetriever {
+  void getAsync(DataProductRetriever&, int index, TaskHolder) final {}
```

Comment:
I assume here is where we would be better off putting delayed fetches of data products, right?
Looks like no other source is doing this though, maybe because all products are accessed in the current setup?